presto-parser/src/main/java/com/facebook/presto/sql/parser/warnings/DefaultParsingWarningCollector.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.parser.warnings;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+public class DefaultParsingWarningCollector
+        implements ParsingWarningCollector
+{
+    private final ImmutableList.Builder<ParsingWarning> builder = ImmutableList.builder();
+
+    @Override
+    public void add(ParsingWarning warning)
```

Comment:
I don't expect parser to be run in multiple threads, but just i case make it synchronized similarly to DefaultWarningCollector. Contention free locks are very cheap.

Reply:
It now adds to the WarningCollector and the default implementation is threadsafe.